### ISSUE TITLE
Calendar is condensed

### ISSUE THREAD

**Issue by MechEng70** [kudrykv/latex-yearly-planner#67]
The output file has a condensed calendar, incorrectly formatted calendar and missing date at the top of the daily pages.

[rm2.mos.default.dailycal.log](https://github.com/kudrykv/latex-yearly-planner/files/10196103/rm2.mos.default.dailycal.log)

[rm2.mos.default.dailycal.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10196105/rm2.mos.default.dailycal.pdf)

[rm2.mos.default.dailycal.out.log](https://github.com/kudrykv/latex-yearly-planner/files/10196109/rm2.mos.default.dailycal.out.log)

Would appreciate any insight in the issue.

Running this on a macbook air.

**Comment by kudrykv** [comment]
Hi @MechEng70!
Can you please put more info on what you're doing? Are you getting a bad planner after the clean clone, or did you do any modifications?

**Comment by MechEng70** [comment]
Absolutely..

I am trying to change the time on each of the daily, to start a 0600 and go to 1800.

Here is my command:

PLANNER_YEAR=2023 PASSES=1 CFG="cfg/rm2.base.yaml,cfg/template_months_on_side.yaml,cfg/rm2.mos.default.dailycal.yaml" NAME="rm2.mos.dailycal.default" ./single.sh

Attached is the one file I changed. (I added .txt to upload it :) )

[rm2.mos.default.dailycal.yaml.txt](https://github.com/kudrykv/latex-yearly-planner/files/10196310/rm2.mos.default.dailycal.yaml.txt)

**Comment by MechEng70** [comment]
did a little digging and there were comments about the xcolor package.. so currently updating all the packages with the TexLive app on MacOS.  Will let you know if it succeeds.

**Comment by MechEng70** [comment]
Did not work as I expected.

Getting 40678 of these types of xcolor errors, not sure if they are related to the condensed calendar.  Have also noticed that the month/quarter bar on the left sometimes when compiled shifts to the right.

```
 ./out/weekly.tex:4527: Package xcolor Error: Undefined color `'.

See the xcolor package documentation for explanation.
Type  H <return>  for immediate help.
```

**Comment by MechEng70** [comment]
Any ideas?

**Comment by kudrykv** [comment]
Hey
Thanks for the wait
Let's try digging step by step

First, did you try compiling whatever was in the repo from the beginning? That would help eliminating a problem of something missing in the system

**Comment by MechEng70** [comment]
no idea why changing just the two have such a profound effect...

I am out of the house for the next 10 hours.  will try to do this when I return or see if I can ssh into my ubuntu server and try it on that platform.  Thanks

**Comment by MechEng70** [comment]
Before pulling down everything, ran the code on the main page and it produced a pdf without any obvious calendar compression.
Using Github desktop, I identified the one file that had the 8 changed to 6 and 20 to 18.  Replaced that file and ran the code:

PLANNER_YEAR=2023 PASSES=1 CFG="cfg/rm2.base.yaml,cfg/template_months_on_side.yaml,cfg/rm2.mos.default.dailycal.yaml" NAME="rm2.mos.dailycal.06001800" ./single.sh

And the calendars are compressed.

![image](https://user-images.githubusercontent.com/11697710/208216756-b9464019-6f41-45f1-88ca-aafb9ec08243.png)

**Comment by kudrykv** [comment]
You need 2 passes for MOS planners. Please, try running again

**Comment by MechEng70** [comment]
Will do in the morning.  Thank you.

**Comment by MechEng70** [comment]
Did run the rm2 code twice and it didn't produce anything different from the image above.  This seems to only happen with the rm2 cfg files.

**Comment by MechEng70** [comment]
Noticed that if the following was added to the rm2.base.yaml:

  colors:
    gray: gray
    lightgray: gray!50

Changed this line:
quarterlyspring: \textcolor{white}{..}

quarterlyspring: \textcolor{white}{.}. <- removed one of the dots.

There are less xcolor errors.
Ran the command twice.  attached the logs and pdf.
[rm2.mos.default.dailycal.log](https://github.com/kudrykv/latex-yearly-planner/files/10252369/rm2.mos.default.dailycal.log)
[rm2.mos.default.dailycal.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10252370/rm2.mos.default.dailycal.pdf)

**Comment by kudrykv** [comment]
What are your versions of latex and pdflatex?
```
$ pdflatex --version
pdfTeX 3.141592653-2.6-1.40.24 (TeX Live 2022)
kpathsea version 6.3.4
Copyright 2022 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.37; using libpng 1.6.37
Compiled with zlib 1.2.11; using zlib 1.2.11
Compiled with xpdf version 4.03
```

I imagine something must be outdated.

---

```
$ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   cfg/rm2.mos.default.dailycal.yaml

$ git diff cfg/rm2.mos.default.dailycal.yaml
diff --git a/cfg/rm2.mos.default.dailycal.yaml b/cfg/rm2.mos.default.dailycal.yaml
index a91297a..f9c9560 100644
--- a/cfg/rm2.mos.default.dailycal.yaml
+++ b/cfg/rm2.mos.default.dailycal.yaml
@@ -2,10 +2,10 @@ calafterschedule: true
 
 layout:
   numbers:
-    dailybottomhour: 8
-    dailytophour: 20
+    dailybottomhour: 6
+    dailytophour: 18
```

That's on `main` branch. And here are PDFs I get for this config entry:
```
2 "cfg/base.yaml,cfg/rm2.base.yaml,cfg/template_months_on_side.yaml,cfg/rm2.mos.default.yaml,cfg/rm2.mos.default.dailycal.yaml"           "rm2.mos.default.dailycal"
```

[rm2.mos.dotted.default.dailycal.2022.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10253545/rm2.mos.dotted.default.dailycal.2022.pdf)
[rm2.mos.dotted.default.dailycal.2023.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10253546/rm2.mos.dotted.default.dailycal.2023.pdf)
[rm2.mos.dotted.default.dailycal.2024.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10253547/rm2.mos.dotted.default.dailycal.2024.pdf)

**Comment by kudrykv** [comment]
Oh. Also, another possible source of issue:
I see you specify the config as `CFG="cfg/rm2.base.yaml,cfg/template_months_on_side.yaml,cfg/rm2.mos.default.dailycal.yaml"`. As the repo works today (which I hope to change in the future ¯\_(ツ)_/¯), is that it takes all the configurations from the list, merging right values to left values. So, as you dropped the `cfg/base.yaml`, you've lost a bunch of configs

**Comment by MechEng70** [comment]
The last comment solved the issue with the calendars.  SW version are the same.

How do I have lines in the notes section of the daily?  and thank you for walking me through this.

**Comment by MechEng70** [comment]
also, where would you put the cfg/base.yaml in the line above?  I have tried it before and after the first config.  the challenge is that now the calendar is not compressed but shifted to the next line.

I will play around with the order today

[rm2.mos.dailycal.2023.pdf](https://github.com/kudrykv/latex-yearly-planner/files/10254260/rm2.mos.dailycal.2023.pdf)

**Comment by MechEng70** [comment]
I have everything working with the following code:
PLANNER_YEAR=2023 PASSES=1 CFG="cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/rm2.mos.default.dailycal.yaml,cfg/rm2.base.yaml" NAME="rm2.mos.dailycal.0618.0627.2023" ./single.sh

The configuration files have is so that the time is from 0600 to 1800 and there are 06 todos and 27 lines for notes.  The only thing I would like to change is the calendar starting on Sunday (american thing :) )

Appreciate all the help.  I have been running it twice and been successful.  Would changing PASSES=1 to PASSES=2 accomplish the same thing?

**Comment by kudrykv** [comment]
> I have been running it twice and been successful. Would changing PASSES=1 to PASSES=2 accomplish the same thing?

Yes. There is a loop, that runs `PASSES` number of times.

> I would like to change is the calendar starting on Sunday

In `base.yaml`, change the `weekstart: 1` to `weekstart: 0` (0 is Sunday, 1 is Monday, 2 is Tuesday, ...)

> Appreciate all the help

❤️

**Comment by kudrykv** [comment]
> How do I have lines in the notes section of the daily?

The dotted/lined is planner-wide, not only the daily pages.
For it, change the `dotted: true` to `dotted: false` in the `base.yaml`

**Comment by kudrykv** [comment]
tl;dr other way of running it. May be not needed.

---

> where would you put the cfg/base.yaml in the line above?

Not sure what you mean. The value to the right overwrites the value to the left. You could even have a single YAML file if you merge all configs into one file. The many configuration files are my embarrassing way to lower the duplication there.

> How do I have lines in the notes section of the daily?

Again, the config of the `main` branch is far from ideal, so bear with me.
Open the `release.sh` and within configurations array, remove or comment _lines_ that you don't need:
![CleanShot 2022-12-19 at 12 44 41](https://user-images.githubusercontent.com/12697803/208408786-8a12a3d3-b9d9-4efc-9dcf-f87827268477.png)

Edit years for which you want to generate the planner:
![CleanShot 2022-12-19 at 12 50 38](https://user-images.githubusercontent.com/12697803/208409767-d1300851-9e49-4692-b6f2-3d07578b1f86.png)

Then, in the same file more lines below, there is another config: dotted/lined, Sunday/Monday, AMPM/2400. Comment out or remove all lines that you don't need. Only one should be left.
![CleanShot 2022-12-19 at 12 48 40](https://user-images.githubusercontent.com/12697803/208409366-1d1a10a9-d632-4ef1-bd2f-9d0134cc5564.png)

Now, run the `./release.sh`. It'll generate the planner for you into the `pile` directory

**Comment by MechEng70** [comment]
Perfect!  Thank you.

**Comment by kudrykv** [comment]
Please reopen if any other issues regarding this occur
